odata-data-aggregation-ext/7 Examples.md

```diff
@@ -1345,6 +1345,117 @@ ancestors(
+Content-Type: application/json
+
+{ "@id": "SalesOrganizations('EMEA Central')" }
+```
+results in `204 No Content`.
+:::
+
+::: example
+Example ##ex: If the parent navigation property contained a referential constraint for the key of the target [OData-CSDL, section 8.5](#ODataCSDL),
+```xml
+<EntityType Name="SalesOrganization">
+  <Key>
+    <PropertyRef Name="ID" />
+  </Key>
+  <Property Name="ID" Type="Edm.String" Nullable="false" />
+  <Property Name="Name" Type="Edm.String" />
+  <Property Name="SuperordinateID" Type="Edm.String" />
+  <NavigationProperty Name="Superordinate"
+                      Type="SalesModel.SalesOrganization">
+    <ReferentialConstraint Property="SuperordinateID"
+                           ReferencedProperty="ID" />
+  </NavigationProperty>
+</EntityType>
+```
+then alternatively the property taking part in the referential constraint [OData-Protocol, section 11.4.9.1](#ODataProtocol) could be changed to EMEA Central:
+```json
+PATCH /service/SalesOrganizations('Switzerland')
+Content-Type: application/json
+
+{ "SuperordinateID": "EMEA Central" }
+```
+:::
+
+If the parent-child relationship between sales organizations is maintained in a separate entity set,
+the entity key may differ from the node identifier property and there can be entities without node identifier. And by using a non-[standard definition of root](#RecursiveHierarchy), even nodes with node identifier can be unreachable from any root, these are called orphans.
+
+::: example
+⚠ Example ##ex: Assume additional `SalesOrganizations` Mars, Phobos and Venus, and that only Sales is a root:
+```xml
+<EntityType Name="SalesOrganizationRelation">
+  <Key>
+    <PropertyRef Name="NodeID" />
+    <PropertyRef Name="SuperordinateID" />
+  </Key>
+  <Property Name="ID" Type="Edm.String" Nullable="false" />
+  <Property Name="SuperordinateID" Type="Edm.String" />
+  <NavigationProperty Name="Superordinate"
+                      Type="SalesModel.SalesOrganization">
+    <ReferentialConstraint Property="SuperordinateID"
+                           ReferencedProperty="ID" />
+  </NavigationProperty>
+</EntityType>
+<EntityType Name="SalesOrganization">
+  <Key>
+    <PropertyRef Name="ID" />
+  </Key>
+  <Property Name="ID" Type="Edm.String" Nullable="false" />
+  <Property Name="Name" Type="Edm.String" />
+  <Property Name="NodeID" Type="Edm.String" />
+  <NavigationProperty Name="Relations"
+                      Type="Collection(SalesModel.SalesOrganizationRelation)">
+    <ReferentialConstraint Property="NodeID"
+                           ReferencedProperty="NodeID" />
+  </NavigationProperty>
+  <Annotation Term="Aggregation.RecursiveHierarchy"
+              Qualifier="MultiParentHierarchy">
+    <Record>
+      <PropertyValue Property="NodeProperty"
+                     PropertyPath="NodeID" />
+      <PropertyValue Property="ParentNavigationProperty"
+                     PropertyPath="Relations/Superordinate" />
```

Comment:
This navigation property path consists of a collection-valued navigation property followed by a single-valued one.